constructr-coordination/src/main/scala/de/heikoseeberger/constructr/coordination/ConsulCoordination.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2015 Heiko Seeberger
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.heikoseeberger.constructr.coordination
+
+import akka.http.scaladsl.client.RequestBuilding.{ Get, Put }
+import akka.http.scaladsl.model.StatusCodes.{ NotFound, OK }
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse, ResponseEntity, Uri }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ ExecutionContext, Future }
+
+final class ConsulCoordination(prefix: String, clusterName: String, host: String, port: Int, send: HttpRequest => Future[HttpResponse])
+    extends Coordination[Coordination.Backend.Consul.type](prefix, clusterName, host, port, send) {
+  import Coordination._
+
+  private val v1Uri = Uri(s"http://$host:$port/v1")
+
+  private val kvUri = Uri(s"http://$host:$port/v1/kv")
+
+  private val baseUri = kvUri.withPath(kvUri.path / "constructr" / prefix / clusterName)
+
+  private val nodesUri = baseUri.withPath(baseUri.path / "nodes")
+
+  private val createSessionUri = v1Uri.withPath(v1Uri.path / "session" / "create")
+
+  override def getNodes[A: AddressSerialization]()(implicit ec: ExecutionContext, mat: Materializer): Future[List[A]] = {
+    def unmarshalNodes(entity: ResponseEntity) = {
+      def toNodes(s: String) = {
+        import rapture.json._
+        import rapture.json.jsonBackends.spray._
+        def jsonToNode(json: Json) = {
```

Comment:
More nitpicks ;-)
These curly braces aren't needed. Move the next line up, just after the equals sign.